### PR TITLE
Load more mobile sessions when scrolling down

### PR DESCRIPTION
**Pull request by Sonje**
### Changes

- decided to keep the update functions within `useMapParams` to keep the logic contained and easily understandable in the context it is used (thus why I didn't extract it)
- created a custom hook that can be reused on scroll end in other components if needed ^_^
- changed mobile session list key from `index` to `session.id` to improve performance on mobile
- added a `clearMobileSessions` reducer to handle residual data messing up with scrolling behaviour and for cleanliness of the code
- removed "Results" from session list header on desktop - there's too much information to keep on one line and it got wrapped. We'd need to make the list wider to incorporate for long session lists with big numbers - on the same account I added "of" for fixed sessions for consistency and to imitate production behaviour - we'll change it if Michael sees it fit.
image
- made sure the refresh button and realtime updates button play nicely with scrolling behaviour by clearing sessions after invoking them.

### Testing notes

- play around with session list scrolling, 
- zoom map in and out while scrolled through the end of the list, 
- copy URL to another tab, 
- test how the scrolling detection reacts with refresh button and map update switch

### CODE REVIEW

File: Procfile.dev [hunk 1]
```diff
@@ -1,3 +1,3 @@
 web: unset PORT && env RUBY_DEBUG_OPEN=true bin/rails server
 js: yarn dev
-sidekiq: bin/sidekiq
+sidekiq: bin/sidekiq
```
Comment (evulins) [inline Procfile.dev:3]: Do we want to add it?

Comment (Sonje) [reply Procfile.dev:3]: I mean, I have it removed locally, but I think that's important when you run backend and frontend locally at the same time

File: app/javascript/react/components/Map/Map.tsx [hunk 2]
```diff
@@ -205,13 +219,51 @@ const Map = () => {
...
+        limit: fetchedSessions,
+      };
+
+      dispatch(fetchMobileSessions({ filters: JSON.stringify(updatedFilters) }))
+        .unwrap()
+        .then(() => {
+          updateLimit(originalLimit);
+          updateFetchedSessions(fetchedSessions);
+        });
+      console.log("fetched sessions first if", fetchedSessions);
```
[the remark this answers is not in the document]
Comment (Sonje) [reply app/javascript/react/components/Map/Map.tsx:238]: yes, thanks

File: app/javascript/react/components/Map/Map.tsx [hunk 3]
```diff
@@ -205,13 +219,51 @@ const Map = () => {
...
+        if (fixedSessionTypeSelected) {
+          dispatch(fetchFixedSessions({ filters })).unwrap();
+        } else {
+          dispatch(fetchMobileSessions({ filters }))
+            .unwrap()
+            .then((response) => {
+              updateFetchedSessions(response.sessions.length);
+            });
+        }
+        console.log("fetched sessions second", fetchedSessions);
```
Comment (evulins) [inline app/javascript/react/components/Map/Map.tsx:251]: The same here

File: app/javascript/react/components/Map/Map.tsx [hunk 4]
```diff
@@ -281,7 +333,81 @@ const Map = () => {
...
+  // }, [
+  //   offset,
+  //   listSessions.length,
+  //   fetchableMobileSessionsCount,
+  //   limit,
+  //   updateOffset,
+  //   dispatch,
+  //   filters,
+  //   updateFetchedSessions,
+  // ]);
```
Comment (evulins) [inline app/javascript/react/components/Map/Map.tsx:409]: Do we need this commented code?

Comment (Sonje) [reply app/javascript/react/components/Map/Map.tsx:409]: nope, this is another leftover -> I thought I removed it during rebase cleanup, I'll remove it

**Review by evulins — APPROVED** [review]
Great job! 💯 🦃 Just minor stuff :)

**Comment by Sonje** [comment]
[Mobile sessions > sessions list: missing design and feature](https://trello.com/c/DwsowEbV/1932-mobile-sessions-sessions-list-missing-design-and-feature)

**Pull request merged**